doc/source/examples/incompressible-flow/3d-taylor-green-vortex/3d-taylor-green-vortex.rst

```diff
@@ -0,0 +1,374 @@
+
+.. code-block:: text
+
+  subsection linear solver
+    subsection fluid dynamics
+      set verbosity               = verbose
+      set method                  = gmres
+      set max iters               = 200
+      set max krylov vectors      = 200
+      set relative residual       = 1e-4
+      set minimum residual        = 1e-12
+      set ilu preconditioner fill               = 0
+      set ilu preconditioner absolute tolerance = 1e-12
+      set ilu preconditioner relative tolerance = 1.00
+    end
+  end
+
+Matrix-free  - Non-linear Solver 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The non-linear solver used in the matrix-free solver is straightforward. We use Newton's method with a tolerance of :math:`10^{-3}`
+
+.. code-block:: text
+
+  subsection non-linear solver
+    subsection fluid dynamics
+      set tolerance      = 1e-3
+      set verbosity      = verbose
+    end
+  end
+
+Matrix-free - Linear Solver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``lethe-fluid-matrix-free`` has significantly more parameters for its linear solver. The new parameters are all related to the geometric multigrid preconditioner that is used by the matrix free algorithm.
+
+.. code-block:: text
+
+  subsection linear solver
+    subsection fluid dynamics
+      set method            = gmres
+      set max iters         = 100
+      set relative residual = 1e-4
+      set minimum residual  = 1e-7
+      set preconditioner    = gcmg
+      set verbosity         = verbos
+      
+      #MG parameters
+      set mg verbosity       = quiet
+      set mg min level       = -1
+      set mg level min cells = 16
+
+      #smoother
+      set mg smoother iterations = 10
+      set mg smoother eig estimation = true
+      
+      # Eigenvalue estimation parameters
+      set eig estimation degree          = 3
+      set eig estimation smoothing range = 5
+      set eig estimation cg n iterations = 20
+      set eig estimation verbosity       = verbose
+
+      #coarse-grid solver
+      set mg coarse grid max iterations     = 2000
+      set mg coarse grid tolerance          = 1e-7
+      set mg coarse grid reduce             = 1e-4
+      set mg coarse grid max krylov vectors = 30
+      set mg coarse grid preconditioner     = ilu
+      set ilu preconditioner fill               = 1
+      set ilu preconditioner absolute tolerance = 1e-10
+      set ilu preconditioner relative tolerance = 1.00
+    end
+  end
+
+We set ``mg verbosity = quiet`` to prevent logging of the multigrid parameters during the simulation. Setting ``mg min level = -1`` ensures that the ``mg level min cells=16`` parameter is used to determine the coarsest level. It is important to ensure that the Taylor-Green vortex has sufficient cell on the coarsest level since periodic boundary conditions are used. Indeed, using a coarsest level with a single cell can lead to a problematic situation where too few degrees of freedom are available on the coarsest level.
```

Comment:
```suggestion
We set ``mg verbosity = quiet`` to prevent logging of the multigrid parameters during the simulation. Setting ``mg min level = -1`` ensures that the ``mg level min cells=16`` parameter is used to determine the coarsest level. It is important to ensure that the Taylor-Green vortex has sufficient cells on the coarsest level since periodic boundary conditions are used. Indeed, using a coarsest level with a single cell can lead to a problematic situation where too few degrees of freedom are available on the coarsest level.
```